### PR TITLE
libusb: add doc output

### PR DESCRIPTION
**Pull request by Qyriad**
## Description of changes

This adds a `withDocs` optional argument to libusb's package definition, which when enabled will build libusb's documentation.

I made it default to true when `buildPlatform.canExecute hostPlatform`, which as far as I could tell is how this is usually done.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] (?) aarch64-linux (built with `{ localSystem.system = "x86_64-linux"; crossSystem.system = "aarch64-linux";`)
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
	- I have not done this at this time due to the mass rebuild
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
	- [ ] Tested basic functionality via `usbutils`, which uses libusb
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by prusnak** [comment]
Please use lib.optionals with a list instead of lib.optional

**Comment by Qyriad** [comment]
@prusnak done. Is this convention written down anywhere? [CONTRIBUTING.md lists both](https://github.com/NixOS/nixpkgs/blob/9699c4d6d13d7dc1c73962bb0ac1be4163fd8831/CONTRIBUTING.md?plain=1#L737).

File: pkgs/development/libraries/libusb1/default.nix [hunk 1]
```diff
@@ -24,9 +26,12 @@ stdenv.mkDerivation rec {
...
   };
 
-  outputs = [ "out" "dev" ];
+  outputs = [ "out" "dev" ] ++ lib.optionals withDocs [ "doc" ];
 
-  nativeBuildInputs = [ pkg-config autoreconfHook ];
+  nativeBuildInputs = [
+    pkg-config
+    autoreconfHook
+  ] ++ lib.optional withDocs doxygen;
```
Comment (prusnak) [inline pkgs/development/libraries/libusb1/default.nix:34]:
```suggestion
  ] ++ lib.optionals withDocs [ doxygen ];
```

Comment (prusnak) [inline pkgs/development/libraries/libusb1/default.nix:34]:
```suggestion
  ] ++ lib.optionals withDocs [ doxygen ];
```

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by prusnak** [comment]
> Is this convention written down anywhere?

I remember reading about it somewhere, but can't remember where.

File: pkgs/development/libraries/libusb1/default.nix (hunk 1 above)
Comment (Qyriad) [reply pkgs/development/libraries/libusb1/default.nix]: Whoops, sorry for missing this one; fixed!

**Review by prusnak — APPROVED** [review]

**Review by prusnak — APPROVED** [review]

**Comment by lheckemann** [comment]
Re optional: https://discourse.nixos.org/t/lib-optional-considered-harmful/7081/5

**Pull request merged**